### ISSUE TITLE
[event] Event.js failed after rework event APIs

### ISSUE THREAD

**Issue by qiaojingx** [01org/zephyr.js#1432]
### Description

 Running [Event.js](https://github.com/01org/zephyr.js/tree/master/samples/tests/Event.js) failed after rework event APIs.  `test_event1` is treated as `test_event` in [Event.js](https://github.com/01org/zephyr.js/tree/master/samples/tests/Event.js).

### Test Code

[Event.js](https://github.com/01org/zephyr.js/tree/master/samples/tests/Event.js);
 
### Steps to Reproduction

 1. `make JS=samples/tests/Event.js`
 2. `make dfu`
### Actual Result

 
![selection_143](https://user-images.githubusercontent.com/19813546/29019580-811c391a-7b91-11e7-8f32-1e59777d9975.png)

### Expected Result

 sample test pass.

### Test Builds

Branch | Commit Id | Target Device | Test Date | Result
---|---|---|---|---
master | f3625a8 | Arduino 101 | Aug 7, 2017 | Fail
master | f3625a8 | FRDM-K64F | Aug 7, 2017 | Fail
master | f3625a8 | Linux | Aug 7, 2017 | Fail
master | e40cd69 | Arduino 101 | Aug 1, 2017 | Pass
master | e40cd69 | FRDM-K64F | Aug 1, 2017 | Pass
master | e40cd69 | Linux | Aug 1, 2017 | Pass

**Comment by cuiyanx** [comment]
I rename the event `test_event1` to `event_test` and remove` -1 `on 34 line, this sample work well.
```
EventName[0]: test_event
EventName[1]: event_test
testListenerNames.length = 2
testListenerCount.length = 2
test_event: handler 1: arg1=1234 arg2=4567
test_event: handler 2: arg1=1234 arg2=4567
event_test: arg1=1000 arg2=1111
test_event: handler 2: arg1=1234 arg2=4567
Listeners for test_event = 1
Listeners for test_event after removeAllListeners = 0
Event test passed
```

**Comment by grgustaf** [comment]
Thanks, I provided a simpler fix in #1435 though.

**Comment by qiaojingx** [comment]
Verified with commit ca7af31 on Arduino101, FRDM-K64F and Linux. This issue is not reproduced.
